### DISCUSSION TITLE
Feat: Need an efficient all document query in RAG

### DISCUSSION THREAD

**Discussion by Shamepoo** [open-webui/open-webui#3923] [General]
**Is your feature request related to a problem? Please describe.**
I noticed that what current all document query in RAG does is to iter all documents(collections), which makes it cannot work when there is a large amount of documents.

**Describe the solution you'd like**
A high performance all document query.

**Describe alternatives you've considered**
Support for other VectorDB, such as PGVector or Milvus

**Additional context**
=

**Comment by alyiox** [comment]
mark it. Any update?

**Comment by RWayne93** [comment]
+1 for pgvector
